go.mod

```diff
@@ -37,6 +37,7 @@ require (
 	github.com/shurcooL/httpfs v0.0.0-20190707220628-8d4bc4ba7749
 	github.com/shurcooL/vfsgen v0.0.0-20200824052919-0d455de96546
 	github.com/stretchr/testify v1.9.0
+	github.com/trivago/tgo v1.0.7
```

Comment:
Not terribly excited to depend on a project that isn't very active (e.g. it doesn't have a go.mod). While I understand that it is stable and proven, I'll look for other options.

Can you also provide an example of something that breaks encoding/json?

Reply:
```
custom_fields:
  customfield_10010: { value: "Monitoring" }
```

Alertmanager parses the nested map as `map[interface{}]interface{}` and `encoding/json` does not support this kind of map. The library ensures that each nested map is casted to `map[string]interface{}`

Reply:
>  I'll look for other options.

https://github.com/json-iterator/go natively supports encoding of `map[interface{}]interface{}`, but I feel there would much less acceptance for a different JSON library.

Reply:
I have no clue, why json-iterator works fine on the lined go test above, but while integrate it inside AM, tests in AM are failed with 

```
error marshalling request body: jira.issue.Fields: json: unsupported type: map[interface {}]interface {}
```

I may give up here, since I don't know why it works on a synthetic environment, but not on AM.

For reference, here is my code (https://github.com/prometheus/alertmanager/commit/cf85bbec731b1530e49daf1ee8a2a58974588b5c)